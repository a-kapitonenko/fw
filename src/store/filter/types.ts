import { Frame } from '../frames/types';

export type Field = {
  name: string;
  value: string;
  checked?: boolean;
};

export type Groups = {
  color: Field[];
  width: Field[];
  noseBridge: Field[];
  shape: Field[];
  material: Field[];
};

export const enum FilterActionTypes {
  FETCH_REQUEST = '@@filter/FETCH_REQUEST',
  FETCH_SUCCESS = '@@filter/FETCH_SUCCESS',
  FETCH_ERROR = '@@filter/FETCH_ERROR',
  CHANGE_CHECKED = '@@filter/CHANGE_CHECKED',
};

export interface IFilterState {
  readonly fetching: boolean;
  readonly groups: Groups;
  readonly errors: string;
  readonly query: Object;
  readonly frames: Frame[];
};
