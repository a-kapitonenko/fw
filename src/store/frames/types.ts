export type Frame = {
  upc: number;
  label: string;
  value: string; 
  img: string;
  compatibility?: boolean;
};

export const enum FramesActionTypes {  
  SUBMIT_START = '@@frames/SUBMIT_START',
  SUBMIT_SUCCESS = '@@frames/SUBMIT_SUCCESS',
  SUBMIT_FAILED = '@@frames/SUBMIT_FAILED',
  FETCH_SIMILAR_FRAMES_START = '@@frames/FETCH_SIMILAR_FRAMES_START',
  FETCH_SIMILAR_FRAMES_SUCCESS = '@@frames/FETCH_SIMILAR_FRAMES_SUCCESS',
  FETCH_SIMILAR_FRAMES_FAILED = '@@frames/FETCH_SIMILAR_FRAMES_FAILED',
  CHECK_FRAME_START = '@@frames/CHECK_FRAME_START',
  CHECK_FRAME_SUCCESS = '@@frames/CHECK_FRAME_SUCCESS',
  CHECK_FRAME_FAILED = '@@frames/CHECK_FRAME_FAILED',
  OPEN = '@@frames/OPEN',
  CLOSE = '@@frames/CLOSE',
  SET_SELECTED_FRAME = '@@frames/SET_SELECTED_FRAME',
  CLEAR_SELECTED_FRAME = '@@frame/CLEAR_SELECTED_FRAME',
  CLEAR_RESULT = '@@frames/CLEAR_RESULT',
};

export interface FramesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly open: boolean;
  readonly selectedFrame: Frame;
  readonly similarFrames: {
    readonly isFetching: boolean;
    readonly errors: string;
    readonly data: Frame[];
  };
};
