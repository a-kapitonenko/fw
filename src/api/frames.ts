import request from './request';
import { Boss } from '../store/order/types';
import { Frame } from '../store/frames/types';
import { onSubmit, getSimilarFrames, checkFrameCompatibility } from '../test/frames';

export const submitRequest = (data: { boss: Boss, frame: Frame }) => request(onSubmit, data);
export const fetchSimilarFramesRequest = (data : Boss) => request(getSimilarFrames, data);
export const checkFrameRequest = (data: Boss) => request(checkFrameCompatibility, data); 
