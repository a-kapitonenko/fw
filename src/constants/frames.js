export const frames = [
  {
    'code': 1,
    'name': '1',
    'url': '1'
  },
  {
    'code': 2,
    'name': '2',
    'url': '2'
  },
  {
    'code': 3,
    'name': '3',
    'url': '3'
  },
  {
    'code': 4,
    'name': '4',
    'url': '4'
  },
  {
    'code': 5,
    'name': '5',
    'url': '5'
  },
  {
    'code': 6,
    'name': '6',
    'url': '6'
  },
  {
    'code': 7,
    'name': '7',
    'url': '7'
  },
  {
    'code': 8,
    'name': '8',
    'url': '8'
  },
  {
    'code': 9,
    'name': '9',
    'url': '9'
  },
  {
    'code': 10,
    'name': '10',
    'url': '10'
  }
];

// export const filterProperties = {
//   color: [
//     { name: 'Black', value: 'black' }, 
//     { name: 'Gold', value: 'gold' }, 
//     { name: 'Brown', value: 'brown'}, 
//     { name: 'Red', value: 'red' }
//   ],
//   fit: {
//     width: [
//       { name: 'Narrow', value: 'narrow' },
//       { name: 'Medium', value: 'medium' },
//       { name: 'Wide', value: 'wide' }
//     ],
//     noseBridge: [
//       { name: 'Standard', value: 'standard' },
//       { name: 'Low bridge fit', value: 'lowBrigdeFit' }
//     ]
//   },
//   shape: [
//     { name: 'Square', value: 'square' },
//     { name: 'Rectangle', value: 'rectangle' },
//     { name: 'Round', value: 'round' },
//     { name: 'Cat-eye', value: 'catEye' }
//   ],
//   material: [
//     { name: 'Acetate', value: 'acetate' },
//     { name: 'Metal', value: 'metal'},
//     { name: 'Mixed material', value: 'mixedMaterial' }
//   ]
// };

export const filterProperties = {
  color: [
    { name: 'Black', value: 'black' }, 
    { name: 'Gold', value: 'gold' }, 
    { name: 'Brown', value: 'brown'}, 
    { name: 'Red', value: 'red' }
  ],
  fit: {
    width: [
      { name: 'Narrow', value: 'narrow' },
      { name: 'Medium', value: 'medium' },
      { name: 'Wide', value: 'wide' }
    ],
    noseBridge: [
      { name: 'Standard', value: 'standard' },
      { name: 'Low bridge fit', value: 'lowBrigdeFit' }
    ]
  },
  shape: [
    { name: 'Square', value: 'square' },
    { name: 'Rectangle', value: 'rectangle' },
    { name: 'Round', value: 'round' },
    { name: 'Cat-eye', value: 'catEye' }
  ],
  material: [
    { name: 'Acetate', value: 'acetate' },
    { name: 'Metal', value: 'metal'},
    { name: 'Mixed material', value: 'mixedMaterial' }
  ]
};