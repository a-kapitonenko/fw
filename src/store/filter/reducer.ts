import { Reducer } from 'redux';
import { IFilterState, FilterActionTypes, Groups, Field } from './types';

const initialGroupsState = {
  color: [],
  width: [],
  noseBridge: [],
  shape: [],
  material: []
}

const initialState: IFilterState = {
  isFetching: false,
  errors: '',
  query: initialGroupsState,
  frames: [],
  groups: {
    isFetching: false,
    errors: '',
    data: <Groups>initialGroupsState,
  }
};

const reducer: Reducer<IFilterState> = (state = initialState, action) => {
  switch (action.type) {
    case FilterActionTypes.FILTERING_START: {
      return { ...state, isFetching: true, errors: '' };
    }
    case FilterActionTypes.FILTERING_SUCCESS: {
      return { ...state, isFetching: false, frames: action.payload };
    }
    case FilterActionTypes.FILTERING_FAILED: {
      return { ...state, isFetching: false, errors: action.payload };
    }
    case FilterActionTypes.FETCH_GROUPS_START: {
      return { ...state, groups: { ...state.groups, isFetching: true, errors: '' } };
    }
    case FilterActionTypes.FETCH_GROUPS_SUCCESS: {
      return { ...state, groups: { ...state.groups, isFetching: false, data: action.payload } };
    }
    case FilterActionTypes.FETCH_GROUPS_FAILED: {
      return { ...state, groups: { ...state.groups, isFetching: true, errors: action.paylaod } };
    }
    case FilterActionTypes.CHANGE_CHECKED: {
      const item = state.groups.data[action.payload.type].map((field: Field) => {
        if (field.name === action.payload.name) {
          return { ...field, checked: action.payload.value };
        }

        return { ...field };
      });

      return {
        ...state,
        groups: {
          ...state.groups,
          data: {
            ...state.groups.data,
            [action.payload.type]: item
          }
        }
      };
    }
    case FilterActionTypes.CLEAR_CHECKED: {
      const groups = <Groups>{};

      for (const group in state.groups.data) {
        groups[group] = state.groups.data[group].map((field: Field) => ({ ...field, checked: false }));
      }

      return { ...state, groups: { ...state.groups, data: groups } };
    }
    case FilterActionTypes.ADD_QUERY: {
      return {
        ...state,
        query: {
          ...state.query,
          [action.payload.type]: [
            ...state.query[action.payload.type],
            action.payload.value
          ]
        }
      };
    }
    case FilterActionTypes.DELETE_QUERY: {
      const index = state.query[action.payload.type].indexOf(action.payload.value);

      return {
        ...state,
        query: {
          ...state.query,
          [action.payload.type]: [
            ...state.query[action.payload.type].slice(0, index),
            ...state.query[action.payload.type].slice(index + 1)
          ]
        }
      };
    }
    case FilterActionTypes.CLEAR_QUERY: {
      return { ...state, query: initialGroupsState };
    }
    default: {
      return state;
    }
  }
};

export { reducer as FilterReducer };
