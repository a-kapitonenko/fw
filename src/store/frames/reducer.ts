import { Reducer } from 'redux';
import { FramesState, FramesActionTypes, Frame } from './types';
import { saveStart, saveSuccess, saveFailed } from '../index';

const initialState: FramesState = {
  isFetching: false,
  errors: '',
  open: false,
  selectedFrame: <Frame>{},
  similarFrames: {
    isFetching: false,
    errors: '',
    data: [],
  },
};

const reducer: Reducer<FramesState> = (state = initialState, action) => {
  switch (action.type) {
    case FramesActionTypes.OPEN: {
      return {
        ...state,
        open: true,
      };
    }
    case FramesActionTypes.CLOSE: {
      return {
        ...state,
        open: false,
      };
    }
    case FramesActionTypes.SUBMIT_START: {
      return saveStart(state);
    }
    case FramesActionTypes.SUBMIT_SUCCESS: {
      return saveSuccess(state);
    }
    case FramesActionTypes.SUBMIT_FAILED: {
      return saveFailed(state, action.payload);
    }
    case FramesActionTypes.FETCH_SIMILAR_FRAMES_START: {
      return {
        ...state,
        similarFrames: saveStart(state.similarFrames),
      };
    }
    case FramesActionTypes.FETCH_SIMILAR_FRAMES_SUCCESS: {
      return {
        ...state,
        similarFrames: saveSuccess(state.similarFrames, 'data', action.payload),
      };
    }
    case FramesActionTypes.FETCH_SIMILAR_FRAMES_FAILED: {
      return {
        ...state,
        similarFrames: saveFailed(state.similarFrames, action.payload),
      };
    }
    case FramesActionTypes.CHECK_FRAME_START: {
      return saveStart(state);
    }
    case FramesActionTypes.CHECK_FRAME_SUCCESS: {
      return saveSuccess(state);
    }
    case FramesActionTypes.CHECK_FRAME_FAILED: {
      return saveFailed(state, action.payload);
    }
    case FramesActionTypes.SET_SELECTED_FRAME: {
      return {
        ...state,
        selectedFrame: action.payload,
      };
    }
    case FramesActionTypes.CLEAR_SELECTED_FRAME: {
      return {
        ...state,
        selectedFrame: <Frame>{},
      };
    }
    case FramesActionTypes.CLEAR_RESULT: {
      return {
        ...initialState,
      };
    }
    default: {
      return state;
    }
  }
};

export { reducer as FramesReducer };
