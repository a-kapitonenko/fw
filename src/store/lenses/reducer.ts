import { Reducer } from 'redux';
import { LensesState, LensesActionTypes } from './types';

const initialState: LensesState = {
  fetching: false,
  list: [],
  errors: '',
}

const reducer: Reducer<LensesState> = (state = initialState, action) => {
  switch (action.type) {
    case LensesActionTypes.FETCH_REQUEST: {
      return { ...state, fetching: true };
    }
    case LensesActionTypes.FETCH_SUCCESS: {
      return { 
        ...state, 
        fetching: false, 
        list: action.payload 
      };
    }
    case LensesActionTypes.FETCH_ERROR: {
      return { 
        ...state, 
        fetching: false, 
        errors: action.payload 
      };
    }
    default: {
      return state
    }
  }
};

export { reducer as LensesReducer }
