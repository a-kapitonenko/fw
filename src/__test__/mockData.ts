import { ApplicationState } from '../store';
import { OculusInfo, Prescription, Boss, Blueprint, Barcode, Errors } from '../store/order/types';
import { Frame, FramesState } from '../store/frames/types';
import { Groups, IFilterState } from '../store/filter/types';
import { Lens } from '../store/lenses/types';
import { initialValue } from '../test/database';
import { RouterState } from 'react-router-redux';
import { ISearchState } from '../store/search/types';

export const errorMessage: string = 'Error message';
export const recommendation: string = 'Based on the prescription, the lens selection will use an 8 base lenses that is 70 MM in diameter. We recommend you look for frames with a strong front curve and a small Frame ED (<60 ED)';
export const message = 'The selected parameters for frame and lens are acceptable for cotout.';
export const checkLensErrorMessage: string = "Selected lens doesn't compatibility with patient prescription. Please, select from the following lenses";
export const checkFrameErrorMessage: string = 'This lens does not accecptable with selected frame.';
export const valuesErrorMessage: string = 'Failed to load order values';

const oculusData: OculusInfo = {
  sphere: '1',
  cyclinder: '2',
  axis: '1',
  addition: '2',
  PDDistance: '1',
  PDNear: '1',
  prism: '1',
};

export const prescription: Prescription = {
  OD: oculusData,
  OS: oculusData,
};

export const lens: Lens = {
  name: 'Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown',
  value: 1,
};

export const lenses: Lens[] = [
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown',
    value: 1
  },
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- Transitions* Signature VII',
    value: 2
  },
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- XTRActive* Gray',
    value: 3
  },
];

export const frame: Frame = {
  upc: 82850553738,
  label: 'FM17116',
  value: '82850553738',
  img: '1.png',
  compatibility: true,
};

export const frames: Frame[] = [
  {
    upc: 82850553738,
    label: 'FM17116',
    value: '82850553738',
    img: '1.png',
    compatibility: true,
  },
  {
    upc: 82850523999,
    label: 'SA17346',
    value: '82850523999',
    img: '2.png',
    compatibility: false,
  },
];

export const groups: Groups = {
  color: [
    { name: 'Black', value: 'black' },
  ],
  width: [
    { name: 'Narrow', value: 'narrow' },
  ],
  noseBridge: [
    { name: 'Standard', value: 'standard' },
  ],
  shape: [
    { name: 'Square', value: 'square' },
  ],
  material: [
    { name: 'Acetate', value: 'acetate' },
  ],
};

export const query: object = {
  color: ['black', 'gold', 'brown'],
  width: ['narrow', 'medium'],
  noseBridge: ['standard'],
  shape: ['square'],
  material: ['acetate'],
};

export const fittingHeight = [
  {
    name: '1',
    value: 1,
  },
  {
    name: '2',
    value: 2,
  },
];

export const blueprint: Blueprint = {
  img: '1.png',
};

export const barcode: Barcode = {
  img: '1.png',
};

export const boss: Boss = {} as Boss;
export const state: ApplicationState = {} as ApplicationState;
export const id: string = '6E8C896C14153342DE34BBF26F69A8135BA548864A8C838F035CB35A593C2399';
export const values: any = initialValue;
export const height: number = 3;

export const filledState: ApplicationState = {
  routing: {} as RouterState,
  frames: {} as FramesState,
  lenses: {
    isFetching: false,
    errors: '',
    lenses: lenses,
  },
  order: {
    isFetching: false,
    errors: <Errors>{},
    redirect: false,
    fittingProperties: [],
    recommendation: '',
    message: '',
    blueprint: <Blueprint>{},
    boss: <Boss>{
      prescription: prescription,
      fittingHeight: 0,
      frame: frame,
      lens: lens,
      barcode: <Barcode>{},
    },
  },
  filter: {} as IFilterState,
  search: {} as ISearchState,
};
