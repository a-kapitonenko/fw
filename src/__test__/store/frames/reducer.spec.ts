import { FramesActionTypes, Frame } from '../../../store/frames/types';
import { FramesReducer } from '../../../store/frames/reducer';
import * as mockData from '../../mockData';

describe('frames reducer', () => {
  it('test handles OPEN', () => {
    const framesAction = {
      type: FramesActionTypes.OPEN,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CLOSE', () => {
    const framesAction = {
      type: FramesActionTypes.CLOSE,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles SUBMIT_START', () => {
    const framesAction = {
      type: FramesActionTypes.SUBMIT_START,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles SUBMIT_SUCCESS', () => {
    const framesAction = {
      type: FramesActionTypes.SUBMIT_SUCCESS,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles SUBMIT_FAILED', () => {
    const framesPayload: string = mockData.errorMessage;

    const framesAction = {
      type: FramesActionTypes.SUBMIT_FAILED,
      payload: framesPayload,
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles FETCH_SIMILAR_FRAMES_START', () => {
    const framesAction = {
      type: FramesActionTypes.FETCH_SIMILAR_FRAMES_START,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles FETCH_SIMILAR_FRAMES_SUCCESS', () => {
    const framesPayload: Frame[] = mockData.frames;

    const framesAction = {
      type: FramesActionTypes.FETCH_SIMILAR_FRAMES_SUCCESS,
      payload: framesPayload,
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles FETCH_SIMILAR_FRAMES_FAILED', () => {
    const framesPayload: string = mockData.errorMessage;

    const framesAction = {
      type: FramesActionTypes.FETCH_SIMILAR_FRAMES_FAILED,
      payload: framesPayload,
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CHECK_FRAME_START', () => {
    const framesAction = {
      type: FramesActionTypes.CHECK_FRAME_START,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CHECK_FRAME_SUCCESS', () => {
    const framesAction = {
      type: FramesActionTypes.CHECK_FRAME_SUCCESS,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CHECK_FRAME_FAILED', () => {
    const framesPayload: string = mockData.errorMessage;

    const framesAction = {
      type: FramesActionTypes.CHECK_FRAME_FAILED,
      payload: framesPayload,
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles SET_SELECTED_FRAME', () => {
    const framesPayload: Frame = mockData.frame;

    const framesAction = {
      type: FramesActionTypes.SET_SELECTED_FRAME,
      payload: framesPayload,
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CLEAR_SELECTED_FRAME', () => {
    const framesAction = {
      type: FramesActionTypes.CLEAR_SELECTED_FRAME,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });

  it('test handles CLEAR_RESULT', () => {
    const framesAction = {
      type: FramesActionTypes.CLEAR_RESULT,
      payload: {},
    };
    
    expect(FramesReducer(undefined, framesAction)).toMatchSnapshot();
  });
});
