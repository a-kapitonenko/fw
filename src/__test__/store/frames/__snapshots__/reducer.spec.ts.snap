// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frames reducer test handles CHECK_FRAME_FAILED 1`] = `
Object {
  "errors": "Error message",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles CHECK_FRAME_START 1`] = `
Object {
  "errors": "",
  "isFetching": true,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles CHECK_FRAME_SUCCESS 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles CLEAR_RESULT 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles CLEAR_SELECTED_FRAME 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles CLOSE 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles FETCH_SIMILAR_FRAMES_FAILED 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "Error message",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles FETCH_SIMILAR_FRAMES_START 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": true,
  },
}
`;

exports[`frames reducer test handles FETCH_SIMILAR_FRAMES_SUCCESS 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [
      Object {
        "compatibility": true,
        "img": "1.png",
        "label": "FM17116",
        "upc": 82850553738,
        "value": "82850553738",
      },
      Object {
        "compatibility": false,
        "img": "2.png",
        "label": "SA17346",
        "upc": 82850523999,
        "value": "82850523999",
      },
    ],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles OPEN 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": true,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles SET_SELECTED_FRAME 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {
    "compatibility": true,
    "img": "1.png",
    "label": "FM17116",
    "upc": 82850553738,
    "value": "82850553738",
  },
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles SUBMIT_FAILED 1`] = `
Object {
  "errors": "Error message",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles SUBMIT_START 1`] = `
Object {
  "errors": "",
  "isFetching": true,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;

exports[`frames reducer test handles SUBMIT_SUCCESS 1`] = `
Object {
  "errors": "",
  "isFetching": false,
  "open": false,
  "selectedFrame": Object {},
  "similarFrames": Object {
    "data": Array [],
    "errors": "",
    "isFetching": false,
  },
}
`;
